tools/clang/test/hlsl_cmds_test/NonUniform.test

```diff
@@ -0,0 +1,56 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
+
+// CHECK: createHandle
+// CHECK: i1 true)
+
+// CHECK: createHandle
+// CHECK: i1 true)
+
+// CHECK: createHandle
+// CHECK: i1 true)
+
+// RUN: %dxc %s /T ps_6_0 /DDX12 /Fo %t.NonUniform.cso
+// RUN: %dxc %t.NonUniform.cso /dumpbin /Qstrip_rootsignature /Fo %t.NonUniformNoRootSig.cso
+// RUN:%dxa  -listparts %t.NonUniformNoRootSig.cso | FileCheck %s --check-prefix=NO_RS
+
+// NO_RS:Part count
+// NO_RS-NOT:RTS0
+
+// RUN: %dxc  %t.NonUniform.cso /dumpbin /extractrootsignature /Fo %t.NonUniformRootSig.cso
+// RUN:%dxa  -listparts %t.NonUniformRootSig.cso | FileCheck %s --check-prefix=RS_PART
+
+// RS_PART:Part count: 1
+// RS_PART:RTS0
+
+// Verify root signature for NonUniform.cso
+// RUN: %dxc  %t.NonUniform.cso /dumpbin /verifyrootsignature %t.NonUniformRootSig.cso
+// RUN: %dxc  %t.NonUniformNoRootSig.cso /dumpbin /verifyrootsignature %t.NonUniformRootSig.cso
+
+// Verify mismatched root signatures fail verification
+
+// RUN: %dxc %S/smoke.hlsl  /D "semantic = SV_Position" /T vs_6_0 /Zi /Qembed_debug /DDX12 /Fo %t.smoke.define.cso
+// RUN: %dxc %t.smoke.define.cso /dumpbin /Qstrip_rootsignature /Fo %t.norootsignature.cso
+// RUN: %dxc %t.smoke.define.cso /dumpbin /extractrootsignature /Fo %t.rootsig.cso
+
+// RUN: not %dxc %t.NonUniformNoRootSig.cso /dumpbin /verifyrootsignature %t.rootsig.cso
+// RUN: not %dxc %t.norootsignature.cso /dumpbin /verifyrootsignature %t.NonUniformRootSig.cso
+
+// RUN: %del %t.NonUniform.cso
```

Comment:
lit requires the Unix shell commands provided by git be on your path, so you could use `rm` instead of a custom delete, but neither should be required since lit deletes temporary files.